Marlin/src/lcd/e3v2/proui/dwin.cpp

```diff
@@ -1541,22 +1549,37 @@ void DWIN_HomingStart() {
 
 void DWIN_HomingDone() {
   HMI_flag.home_flag = false;
-  dwin_zoffset = TERN0(HAS_BED_PROBE, probe.offset.z);
+  #if ENABLED(MESH_BED_LEVELING) && EITHER(BABYSTEP_ZPROBE_OFFSET, JUST_BABYSTEP)
+    planner.synchronize();
+    babystep.add_mm(Z_AXIS, HMI_data.ManualZOffset);
+  #endif
   if (HMI_flag.abort_action) DWIN_Print_Aborted(); else HMI_ReturnScreen();
 }
 
 void DWIN_LevelingStart() {
-  #if HAS_ONESTEP_LEVELING
+  #if HAS_BED_PROBE
```

Comment:
The presence of a probe is no guarantee that `G29` is also available. The `HAS_ONESTEP_LEVELING` flag is meant to indicate that there is a leveling system enabled and it can be done without the user having to do a paper-test at each point and click to continue. In other words, just a simple `G29` is all you need to issue and it will do a complete probe of the bed without any further user interaction.